### PR TITLE
[WFLY-13991]: Keep corrupted journal files in Artemis.

### PR DESCRIPTION
**Pull request by ehsavoie**
Issue: https://issues.redhat.com/browse/WFLY-13991

### CODE REVIEW

**Review by bstansberry — APPROVED** [review]

File: messaging/WFLY-13991_keep_corrupted_journal_files.adoc [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+:email:             your.email@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:          messaging,jms
+:idseparator:       -
+
+== Overview
+
+
+When a corrupted artemis journal file is seen Artemis shuts down EAP, but before it does that it deletes the corrupted file. This means that no after the event analysis/investigation can be done as the corrupted journal file is gone.
```
Comment (bstansberry) [inline messaging/WFLY-13991_keep_corrupted_journal_files.adoc:12]: s/EAP/WildFly

File: messaging/WFLY-13991_keep_corrupted_journal_files.adoc [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+=== Other Interested Projects
+
+== Requirements
+
+* The journalMaxAtticFiles parameter of ActiveMQ Artemis configuration must be configurable via Wildfly management model.
+
+== Implementation Plan
+
+The `journal-max-attic-files` attribute of type INT with a default value of '10' will be added to messaging server resource.
```
Comment (marekkopecky) [inline messaging/WFLY-13991_keep_corrupted_journal_files.adoc:49]: Is it possible to set infinite here? Eg by setting `0` value?

Comment (ehsavoie) [reply messaging/WFLY-13991_keep_corrupted_journal_files.adoc:49]: No
0 means no file stored

File: messaging/WFLY-13991_keep_corrupted_journal_files.adoc [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+=== Affected Projects or Components
+
+* WildFly
+* ActiveMQ Artemis (finished)
+
+=== Other Interested Projects
+
+== Requirements
+
+* The journalMaxAtticFiles parameter of ActiveMQ Artemis configuration must be configurable via Wildfly management model.
```
Comment (marekkopecky) [inline messaging/WFLY-13991_keep_corrupted_journal_files.adoc:45]: Can you please describe requirements for this parameter? Ok, it is INT, default value is 10. I also read something in documentation and implementation PR, but implementation is prepared based on the requirements from this document. Btw, what is attic folder?

Comment (ehsavoie) [reply messaging/WFLY-13991_keep_corrupted_journal_files.adoc:45]: A folder called attic inside the journal folder. This is Artemis we are talking about

**Review by marekkopecky — APPROVED** [review]
@ehsavoie Thank you for updates, I'm approving this PR

**Pull request merged**